### PR TITLE
Update activity status on failure MySqlDataReader.

### PR DESCRIPTION
**Pull request by qq362220083**
When MySqlDataReader has exception, the activity status needs to be updated.

### CODE REVIEW

File: src/MySqlConnector/Utilities/ActivitySourceHelper.cs [hunk 1]
```diff
@@ -36,7 +36,13 @@ internal static class ActivitySourceHelper
 		return activity;
 	}
 
-	public static void SetSuccess(this Activity activity) => activity.SetTag(StatusCodeTagName, "OK");
+	public static void SetSuccess(this Activity activity)
+	{
+		if (activity.Duration == TimeSpan.Zero)
```
Comment (bgrainger) [inline src/MySqlConnector/Utilities/ActivitySourceHelper.cs:41]: What's the purpose of this change?

File: src/MySqlConnector/MySqlDataReader.cs [hunk 2]
```diff
@@ -474,8 +474,13 @@ internal static async Task<MySqlDataReader> CreateAsync(CommandListPosition comm
...
 		}
-		catch (Exception)
+		catch (Exception ex)
 		{
+			if (activity is { IsAllDataRequested: true })
+			{
+				activity.SetException(ex);
+				activity.Stop();
+			}
 			dataReader.Dispose();
```
Comment (bgrainger) [inline src/MySqlConnector/MySqlDataReader.cs:484]: `MySqlDataReader.DisposeAsync` will still Stop the activity and set it to successful. Seems like that logic ought to change to if `if (Activity.Status == ActivityStatusCode.Unset)` to avoid overwriting the status if it's already an error.

**Comment by bgrainger** [comment]
Thanks! This is definitely an omission in MySqlConnector's activity-handling code; thanks for fixing it. Left just a couple of comments about state management in the face of tracking this exception.

File: src/MySqlConnector/Utilities/ActivitySourceHelper.cs (hunk 1 above)
Comment (bgrainger) [reply src/MySqlConnector/Utilities/ActivitySourceHelper.cs:41]: Oh, is this a workaround for a lack of an `IsStopped` property (https://github.com/dotnet/runtime/issues/63353) to check if the Activity is still running? A comment would be helpful here, but I think checking `Status` would be clearer.

Comment (qq362220083) [reply src/MySqlConnector/Utilities/ActivitySourceHelper.cs:41]: uh, activity only supports `Status` property in .net 6 and above.

Here is the resolve to set the Exception first, and then set the Success to cause the state to be overwritten.

set the Exception should have the highest priority.

Comment (bgrainger) [reply src/MySqlConnector/Utilities/ActivitySourceHelper.cs:49]: Thanks; I hadn't checked the docs thoroughly to see where the API was available.

**Comment by bgrainger** [comment]
Thanks for the contribution! This is included in [2.1.9](https://github.com/mysql-net/MySqlConnector/releases/tag/2.1.9).
